### PR TITLE
Implement safe int counter vec as `IntCounterVec<const N:usize>`

### PR DESCRIPTION
**Pull request by guilload**
### Description
Per title.

### How was this PR tested?
The compiler is doing all the work :)

### CODE REVIEW

**Review by fmassot — APPROVED** [review]
Nice :)

File: quickwit/quickwit-common/src/metrics.rs [hunk 1]
```diff
@@ -19,29 +19,35 @@
...
+    IntGauge, IntGaugeVec,
 };
 
+pub struct IntCounterVec<const N: usize> {
+    underlying: PrometheusIntCounterVec,
+}
+
+impl<const N: usize> IntCounterVec<N> {
+    pub fn new(name: &str, description: &str, namespace: &str, labels: [&str; N]) -> Self {
+        let counter_opts = Opts::new(name, description).namespace(namespace);
```
Comment (fulmicoton) [inline quickwit/quickwit-common/src/metrics.rs:32]: The constructor registers the Counter. Despite the similar naming, that's a very different behavior from the prom counters. 

Can we put the code for this constructor in a `new_counter_vec` function.

```
pub fn new_counter_vec<const N: usize>(...) -> IntCounterVec<N> {
...
}
```

**Review by fulmicoton — APPROVED** [review]

**Comment by fulmicoton** [comment]
Ah cool. This will save us time and trouble.

File: quickwit/quickwit-common/src/metrics.rs (hunk 1 above)
Comment (guilload) [reply quickwit/quickwit-common/src/metrics.rs]: Ok.

**Pull request merged**